server/resources/Maps.py

```diff
@@ -114,25 +117,38 @@ def post(self):
 
         # store original indices in array
         index_arr = [i for i in range(len(arr))]
+        init_length = self.calculate_path_length(index_arr, matrix)
```

Comment:
You should move this logic into a separate class and file so you can play with different algorithms

Reply:
honestly it doesn't run that slowly from my tests even with the longest time algorithm but maybe Ill do that yea. It was about 5-7 seconds for 9 points.